### PR TITLE
Fix some memory issues

### PR DESCRIPTION
**Pull request by mattcaswell**
Fix two memory issues detected by #21668.

### CODE REVIEW

**Review by t8m — APPROVED** [review]

**Comment by mattcaswell** [comment]
> Are these master branch only problems?

Yes

File: ssl/record/rec_layer_s3.c [hunk 1]
```diff
@@ -1325,7 +1325,6 @@ int ssl_set_new_record_layer(SSL_CONNECTION *s, int version,
                 next = BIO_new(BIO_s_mem());
 
             if (next == NULL) {
-                BIO_free(prev);
```
Comment (tom-cosgrove-arm) [inline ssl/record/rec_layer_s3.c:1328]: Do we need to keep the `BIO_free(prev)` at (new) line 1366?

Comment (tom-cosgrove-arm) [reply ssl/record/rec_layer_s3.c:1328]: Ah cool, yes, makes sense

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Comment by hlandau** [comment]
Merged to master. Thank you.

**Pull request closed without merging**